microservices/recordings/src/Recording/Encoder.php

```diff
@@ -109,117 +119,149 @@ public function processAction()
             }
 
             // Get Call-Id hash id
-            $hashid = substr(md5($callid),0,8);
+            $hashid = substr(md5($callid), 0, 8);
 
             // Get callid from file
             $this->logger->info(sprintf("[Recordings][%s] Checking file %s\n", $hashid, $file));
 
-            // Look if the converstation with that id has ended
-            /** @var TrunksCdrInterface $kamAccCdr */
-            $kamAccCdr = $this->trunksCdrRepository->findOneByCallid($callid);
-            if ($kamAccCdr) {
-                $type = 'ddi';
-                if ($kamAccCdr->getXcallid()) {
-                    // If call first leg, caller is who activated the recording
-                    $recorder = $kamAccCdr->getCaller();
-                } else {
-                    // If call second leg, callee is who activated the recording
-                    $recorder = $kamAccCdr->getCallee();
-                }
-            } else {
-                $kamAccCdr = $this->usersCdrRepository->findOneByCallid($callid);
-                if ($kamAccCdr) {
-                    $type = 'ondemand';
-                    if ($kamAccCdr->getXcallid()) {
-                        // If call second leg, callee is who activated the recording
-                        $recorder = $kamAccCdr->getCallee();
-                    } else {
+            // Look if the conversation with that id has ended
+            $kamAccCdrs = [];
+            $kamAccCdrs = array_merge($kamAccCdrs, $this->trunksCdrRepository->findBy(['callid' => $callid]));
+            if (empty($kamAccCdrs)) {
+                $kamAccCdrs = array_merge($kamAccCdrs, $this->usersCdrRepository->findBy(['callid' => $callid]));
```

Comment:
¿Could you move this into the repository? Just like the one on trunksCdrRepository